drivers/spi/spi_ll_stm32.h

```diff
@@ -25,57 +25,126 @@ struct spi_stm32_data {
 
 static inline u32_t ll_func_TX_is_empty(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	return LL_SPI_IsActiveFlag_TXP(spi);
+#else
 	return LL_SPI_IsActiveFlag_TXE(spi);
+#endif
 }
 
 static inline u32_t ll_func_RX_is_not_empty(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	return LL_SPI_IsActiveFlag_RXP(spi);
+#else
 	return LL_SPI_IsActiveFlag_RXNE(spi);
+#endif
 }
 
 static inline void ll_func_enable_IT_TX_empty(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_EnableIT_TXP(spi);
+#else
 	LL_SPI_EnableIT_TXE(spi);
+#endif
 }
 
 static inline void ll_func_enable_IT_RX_not_empty(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_EnableIT_RXP(spi);
+#else
 	LL_SPI_EnableIT_RXNE(spi);
+#endif
 }
 
 static inline void ll_func_enable_IT_errors(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_EnableIT_UDR(spi);
+	LL_SPI_EnableIT_OVR(spi);
+	LL_SPI_EnableIT_CRCERR(spi);
+	LL_SPI_EnableIT_FRE(spi);
+	LL_SPI_EnableIT_MODF(spi);
+#else
 	LL_SPI_EnableIT_ERR(spi);
+#endif
 }
 
 static inline void ll_func_disable_IT_TX_empty(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_TXP(spi);
+#else
 	LL_SPI_DisableIT_TXE(spi);
+#endif
 }
 
 static inline void ll_func_disable_IT_RX_not_empty(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_RXP(spi);
+#else
 	LL_SPI_DisableIT_RXNE(spi);
+#endif
 }
 
 static inline void ll_func_disable_IT_errors(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_UDR(spi);
+	LL_SPI_DisableIT_OVR(spi);
+	LL_SPI_DisableIT_CRCERR(spi);
+	LL_SPI_DisableIT_FRE(spi);
+	LL_SPI_DisableIT_MODF(spi);
+#else
 	LL_SPI_DisableIT_ERR(spi);
+#endif
 }
 
 static inline u32_t ll_func_SPI_is_busy(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	return (!LL_SPI_IsActiveFlag_MODF(spi) &&
+		!LL_SPI_IsActiveFlag_TXC(spi));
+#else
 	return LL_SPI_IsActiveFlag_BSY(spi);
+#endif
 }
 
 static inline void ll_func_set_FIFO_threshold_one_data(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_SetFIFOThreshold(spi, LL_SPI_FIFO_TH_01DATA);
```

Comment:
In case of other SPI FIFO IPs, threshold is set to 8-bit. Does 01data matches the same value?
In that case we can rename function as  `ll_func_set_fifo_threshold_8bit`